### DISCUSSION TITLE
Struggling with next/navigation in next13

### DISCUSSION THREAD

**Discussion by ellisza** [vercel/next.js#55121] [App Router]
I have an application I am building in Next 13. I am using useRouter from next/navigation and Link with href to move the user through the site. When developing locally nothing appears wrong. Once I built my app and deployed it on Vercel, the router will only push once, then after that anything with `useRouter` or `<Link href >` does not work. Once I refresh the page, the useRouter/Link works again.

Has anyone experienced anything similar? If you are interested in helping me, I can show you my site and let you see for yourself. Thank you.

**Comment by ellisza** [comment]
Also I should say I have triple checked that I have 'use client'; in every file that uses the next/navigation.

**Comment by icyJoseph** [comment]
Do you have a repository to take a look at? How are you using the router.push method?
